src/options/views/tab-installed.vue

```diff
@@ -378,6 +380,39 @@ export default {
         if (step) await new Promise(resolve => setTimeout(resolve));
       }
     },
+    performSearch(scripts) {
+      let searchRE;
+      let searchError;
+      const [, expr, flags] = this.search.match(/^\/(.+?)\/(\w*)$|$/);
+      if (expr) {
+        try {
+          searchRE = new RegExp(expr, `i${flags.replace('i', '')}`);
+        } catch (err) {
+          searchError = err.message;
+        }
+      }
+      this.searchError = searchError || null;
+      const lowerSearch = !searchRE && (this.search || '').toLowerCase();
+      scripts.forEach(({ $cache }) => {
+        $cache.show = $cache.search.some((item, i, arr) => {
+          if (typeof item !== 'string') {
+            item = item[0].toLowerCase();
```

Comment:
Since the array is always turned into a string on first search, why do we keep the array at first?

Reply:
To avoid the unnecessary lowercasing of all the strings during initialization, which takes time and stresses garbage collector. It's performed now on demand - only when the user actually searches for something. And since it's enough for a first match to succeed, the unused values won't be lowercased unnecessarily.

Reply:
> And since it's enough for a first match to succeed, the unused values won't be lowercased unnecessarily.

Then we can build `$cache.search` on first search, even without the need to build an unnecessary array.

Reply:
The wrapper array (or another method of postponing the unnecessary lowercasing) will still be required. Of course I can move the code that creates $cache.search into `performSearch` but I don't think it's a better way.

Reply:
To illustrate what happens here's the initial `search`:
```
['Foo']
['Bar']
```
* When the user types `f` the first item will be unwrapped and lowercased to `'foo'` string - and stop because `f` matches `foo`.
* When the user continues to type `fu` the first item will be checked unsuccessfully and the second item will be unwrapped and lowercased.
* The process will continue until the last item is processed, which is the script code.

Reply:
I can understand it now. But is it worth adding such complexity just to avoid joining and lowercasing a few, to be exact, 6 at most, one-line strings? And there may likely be empty strings, wasting time to be wrapped and unwrapped.

Reply:
Not for these 6 strings specifically because the difference in performance is almost non-existent, but to support the big `code` added afterwards. Also, wrapping into an array is negligible compared to lowercasing especially if VM will switch to `toLocaleLowerCase` to support unicode fully, which is very slow. As for "complexity" there are only a few statements/calls added which doesn't really seem complex to me.

Reply:
...but I'll remove the modifications I've made in filter/map, as there are no noticeable gains.